File: rust/crates/sift_cli/src/cli/export.rs
use clap::ValueEnum;
use sift_rs::exports::v1::ExportOutputFormat;

#[derive(Debug, Copy, Clone, ValueEnum)]
pub enum Format {
    Csv,
    Sun,
}

impl From<Format> for ExportOutputFormat {
    fn from(val: Format) -> Self {
        match val {
            Format::Csv => ExportOutputFormat::Csv,
            Format::Sun => ExportOutputFormat::Sun,
        }
    }
}

File: python/lib/sift_client/sift_types/export.py
from __future__ import annotations

from enum import Enum

from sift.exports.v1.exports_pb2 import ExportOutputFormat as ExportOutputFormatProto


class ExportOutputFormat(Enum):
    """Supported output formats for data exports.

    Attributes:
        CSV: Comma-separated values format.
        SUN: Sun (WinPlot) format (not used in certain environments).
        PARQUET: Apache Parquet columnar storage format.
    """

    CSV = ExportOutputFormatProto.EXPORT_OUTPUT_FORMAT_CSV
    SUN = ExportOutputFormatProto.EXPORT_OUTPUT_FORMAT_SUN
    PARQUET = ExportOutputFormatProto.EXPORT_OUTPUT_FORMAT_PARQUET
